stdlib/task/src/main/java/org/ballerinalang/stdlib/task/utils/TaskExecutor.java

```diff
@@ -40,9 +38,17 @@ public static void executeFunction(ServiceInformation serviceInformation) {
     }
 
     private static Object[] getParameterList(AttachedFunction function, ServiceInformation serviceInformation) {
-        if (function.type.paramTypes.length > 0 && Objects.nonNull(serviceInformation.getAttachment())) {
-            return new Object[]{serviceInformation.getAttachment(), Boolean.TRUE};
+        Object[] attachments = serviceInformation.getAttachment();
+        int numberOfParameters = function.type.paramTypes.length;
+        Object[] parameters = null;
+        if (numberOfParameters == attachments.length) {
```

Comment:
How does this behave in the error cases? If the # of onTrigger resource param count and the # of attach() function params are different, null will be returned. 

If the above is validated during the compile-time, then this condition is not necessary. WDYT?

Reply:
Ideally, this should be validated in the compile time, but unfortunately, we can't write compiler plugin for this, just yet. (See #14148)

Therefore this is added. But since this is invoked from a resource function, we cannot return any error here. Is there a way to reflect this error scenario?